keyboards/mechwild/sugarglider/sugarglider.c

```diff
@@ -0,0 +1,441 @@
+// Copyright 2023 Kyle McCreery
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "sugarglider.h"
+
+#ifndef GLIDEPOINT_DPI_OPTIONS
+#    define GLIDEPOINT_DPI_OPTIONS \
+        { 400, 800, 1200, 1600, 2000, 2400, 2800, 3200, 3600, 4000 }
+#    ifndef GLIDEPOINT_DPI_DEFAULT
+#        define GLIDEPOINT_DPI_DEFAULT 1
+#    endif
+#endif
+#ifndef GLIDEPOINT_DPI_DEFAULT
+#    define GLIDEPOINT_DPI_DEFAULT 1
+#endif
+
+keyboard_config_t keyboard_config;
+uint16_t          dpi_array[] = GLIDEPOINT_DPI_OPTIONS;
+#define DPI_OPTION_SIZE (sizeof(dpi_array) / sizeof(uint16_t))
+
+void board_init(void) {
+    // B9 is configured as I2C1_SDA in the board file; that function must be
+    // disabled before using B7 as I2C1_SDA.
+    setPinInputHigh(B9);
+    setPinOutput(B12);
+    setPinOutput(B13);
+    setPinOutput(B14);
+    writePinLow(B12);
+    writePinLow(B13);
+    writePinLow(B14);
+    setPinOutput(C13);
+}
+
+#ifdef DYNAMIC_TAPPING_TERM_ENABLE
+void tap_modify(int change_value, bool tap_status) {
+    if (keyboard_config.dt_term_config < 0) {
+        keyboard_config.dt_term_config *= -1;
+    }
+
+    keyboard_config.dt_term_config += change_value;
+
+    if (tap_status == false ) {
+        keyboard_config.dt_term_config *= -1;
+        g_tapping_term = 0;
+    } else {
+        g_tapping_term = keyboard_config.dt_term_config;
+    }
+    eeconfig_update_kb(keyboard_config.raw);
+}
+
+void tap_toggle(void) {
+    keyboard_config.dt_term_config *= -1;
+    if (keyboard_config.dt_term_config > 0) {
+        g_tapping_term = keyboard_config.dt_term_config;
+    } else {
+        g_tapping_term = 0;
+    }
+    eeconfig_update_kb(keyboard_config.raw);
+}
+#endif
+
+#ifdef DIP_SWITCH_ENABLE
+bool dip_switch_update_kb(uint8_t index, bool active) {
+    if (!dip_switch_update_user(index, active)) { return false; }
+    switch (index) {
+        case 0:
+            if(active) { tap_code(KC_CAPS); }
```

Comment:
if you're using the dip switch to set caps lock, then this may make more sense:
```suggestion
            if(active != host_keyboard_led_state().caps_lock) { tap_code(KC_CAPS); }
```

Eg, if the active state doesn't equal the caps state, tap caps, to make it align.